lib/checks/disks.go

```diff
@@ -0,0 +1,119 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checks
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/gravity/lib/defaults"
+	"github.com/gravitational/gravity/lib/network/validation/proto"
+	"github.com/gravitational/gravity/lib/state"
+
+	"github.com/gravitational/trace"
+)
+
+// checkEtcdDisk makes sure that the disk used for etcd wal satisfies
+// performance requirements.
+func (r *checker) checkEtcdDisk(ctx context.Context, server Server) error {
+	// Test file should reside where etcd data will be.
+	testPath := state.InEtcdDir(server.ServerInfo.StateDir, testFile)
+	res, err := r.Remote.CheckDisks(ctx, server.AdvertiseIP, fioEtcdJob(testPath))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	log.Debugf("Server %v disk test results: %s.", server.Hostname, res.String())
+	if len(res.Jobs) != 1 {
+		return trace.BadParameter("expected 1 job result: %v", res)
+	}
+	iops := res.Jobs[0].GetWriteIOPS()
+	latency := res.Jobs[0].GetFsyncLatency()
+	if iops < EtcdMinWriteIOPS || latency > EtcdMaxFsyncLatencyMs {
+		return trace.BadParameter(formatEtcdMessage(server, testPath, iops, latency))
+	}
+	log.Infof("Server %v passed etcd disk check, has %v sequential write iops and %vms fsync latency.",
+		server.Hostname, iops, latency)
+	return nil
+}
+
+// fioEtcdJob constructs a request to check etcd disk performance.
+func fioEtcdJob(filename string) *proto.CheckDisksRequest {
+	spec := &proto.FioJobSpec{
+		Name: "etcd",
+		// perform sequential writes
+		ReadWrite: "write",
+		// use write() syscall for writes
+		IoEngine: "sync",
+		// sync every data write to disk
+		Fdatasync: true,
+		// test file, should reside where etcd WAL will be
+		Filename: filename,
+		// average block size written by etcd
+		BlockSize: "2300",
+		// total size of the test file
+		Size_: "22m",
+		// limit total test runtime
+		Runtime: proto.DurationProto(defaults.DiskTestDuration),
+	}
+	return &proto.CheckDisksRequest{
+		Jobs: []*proto.FioJobSpec{spec},
+	}
+}
+
+// formatEtcdMessage returns appropritate formatted error message based
+// on the etcd disk performance test results.
+func formatEtcdMessage(server Server, testPath string, iops float64, latency int64) string {
+	var errors []string
+	if iops < EtcdMinWriteIOPS {
+		errors = append(errors, fmt.Sprintf("  * Low sequential write IOPS of %v, required minimum is %v.",
+			iops, EtcdMinWriteIOPS))
+	}
+	if latency > EtcdMaxFsyncLatencyMs {
+		errors = append(errors, fmt.Sprintf("  * High fsync latency of %vms, required maximum is %vms.",
+			latency, EtcdMaxFsyncLatencyMs))
+	}
+	return fmt.Sprintf(
+		`It looks like on node %v etcd data resides on a disk (%v) that does not satisfy recommended performance requirements:
+%v
+For optimal performance, please make sure that the directory with etcd data resides on the device that meets the aforementioned
```

Comment:
So the way this error is worded, to me sounds like a warning. However, looking at the code, to me this looks like it will return an error.

I would consider simplifying the message / making it consistent with other checks, such as https://github.com/gravitational/gravity/blob/23989ef97ed7fcc65d0aa560b4c008e957c70b87/lib/checks/checks.go#L455-L458

For example:
`server %q disk IOPS on %q/s is lower than required %v", server.Hostname, filepath.Dir(testPath), strings.Join(errors, "\n")` And then if it formats ok return an aggregate error with the multiple failures for IOPS and latency.